### PR TITLE
Re-arrange directory structure and docs for GKE to support multiple Providers

### PR DESCRIPTION
**Pull request by weastel**
Signed-off-by: Drumil Patel <drumilpatel720@gmail.com>

This pull request introduces the following changes :

- Move PROJECT_ID to GKE_PROJECT_ID.
- Filename changes to make it more specific to gke(i.e cluster.yaml => cluster_gke.yaml, nodepools.yaml =>nodes_gke.yaml)
- `nodepools` to `nodes` in `\infra` to make it more general.
- Add separate documentation for gke(prombench/docs/gke.md)
- Make prombench documentation more general

### CODE REVIEW

File: prombench/manifests/prombench/nodepools.yaml
Comment (geekodour) [inline prombench/manifests/prombench/nodepools.yaml:2]: We want to move this file aswell to the new file name, no? don't forget to update the same in the Makefile :)

File: prombench/Makefile
Comment (geekodour) [inline prombench/Makefile]: I think you'd need to change `noodpool` to `node`? :thinking:

File: funcbench/manifests/cluster.yaml
Comment (geekodour) [inline funcbench/manifests/cluster.yaml:1]: It'll be nice to update the filename for `cluster.yaml` for funcbench aswell :)

**Review by geekodour — COMMENTED** [review]
Nice! This will make the #384 easier to follow. :)

Added a few small suggestions. Once this merged we must also make changes to the workflow files at : https://github.com/prometheus/prometheus/tree/master/.github/workflows

**Comment by geekodour** [comment]
Another thing I would expect to be in this PR is the creation of docs directory and moving the GKE specific docs to `/docs/gke.md`, link to it from the prombench README. So the `/prometheus/README.md` will become significantly smaller.

File: prombench/README.md
Comment (weastel) [inline prombench/README.md]: @geekodour Can you please suggest me better heading and description for this.

**Review by geekodour — COMMENTED** [review]
Thanks for adding the changes and the description changes :) :sparkles: 

Added some suggestion, ~next thing to do with this PR is to update the commands in the funcbench Makefile as it also infra cli~.

File: prombench/Makefile
Comment (geekodour) [inline prombench/Makefile:10]: We should use the `PROVIDER` makefile variable for both funchbench and prombench Makefiles. ( Currently can be need in #384 Makefile for prombench)

File: .github/workflows/grafana_dashboard.yaml
Comment (geekodour) [inline .github/workflows/grafana_dashboard.yaml]:
```suggestion
        AUTH_FILE: ${{ secrets.TEST_INFRA_PROVIDER_AUTH }}
```

**Review by geekodour — CHANGES_REQUESTED** [review]

**Review by geekodour — COMMENTED** [review]
small doc changes, I tested this. will merge once changes are addressed :)

**Comment by geekodour** [comment]
@krasi-georgiev can you take a final look before merging? :)

**Comment by geekodour** [comment]
@weastel you should make a PR to update the prometheus/prometheus github action workflow files aswell that will be affected by changes in this PR.

**Comment by geekodour** [comment]
@krasi-georgiev I merged this as it was blocking few other PRs. Please still take a look at it, if we need to change anything back.

@weastel thanks for this! :)

**Pull request merged**
